cloud/ovirt/ovirt_storage_domains_facts.py

```
@@ -0,0 +1,95 @@
+description:
+    - "Retrieve facts about one or more oVirt storage domains."
+notes:
+    - "This module creates a new top-level C(ovirt_storage_domains) fact, which
+       contains a list of storage domains."
+options:
+    pattern:
+      description:
+        - "Search term which is accepted by oVirt search backend."
+        - "For example to search storage domain X from datacenter Y use following pattern:
+           name=X and datacenter=Y"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all storage domains which names start with C(data) and
+# belong to data center C(west):
+- ovirt_storage_domains_facts:
+    pattern: name=data* and datacenter=west
+- debug:
+    var: ovirt_storage_domains
+'''
+
+RETURN = '''
+ovirt_storage_domains:
+    description: "List of dictionaries describing the storage domains. Storage_domain attribues are mapped to dictionary keys,
+                  all storage domains attributes can be found at following url: https://ovirt.example.com/ovirt-engine/api/model#types/storage_domain."
+    returned: On success.
+    type: list
+'''
+
+
+def main():
+    argument_spec = ovirt_full_argument_spec(
+        pattern=dict(default='', required=False),
+    )
+    module = AnsibleModule(argument_spec)
+    check_sdk(module)
+
+    try:
+        connection = create_connection(module.params.pop('auth'))
+        storage_domains_service = connection.system_service().storage_domains_service()
+        storage_domains = storage_domains_service.list(search=module.params['pattern'])
+        module.exit_json(
+            changed=False,
+            ansible_facts=dict(
+                ovirt_storage_domains=[
+                    get_dict_of_struct(c) for c in storage_domains
+                ],
+            ),
+        )
+    except Exception as e:
+        module.fail_json(msg=str(e))
```

Same here, please surface the full exception.